### ISSUE TITLE
Windows10/Phyton3/Pycharm/2pdata Images must really be same size for register_translation demo_pipeline

### ISSUE THREAD

**Issue by Rack95** [flatironinstitute/CaImAn#344]
For better support, please use the template below to submit your issue.

Sometimes errors while running CNMF occur during parallel processing which prevents the log to provide a meaningful error message.
Please reproduce your error with setting `dview=None`.

If you need to upgrade CaImAn follow the instructions given in the [wiki](https://github.com/flatironinstitute/CaImAn/wiki/Updating-CaImAn).

- **Tell us about your operating system (Linux/macOS/Windows), Python version (2.7/3.x), and working environment (Spyder/Jupyter Notebook/other)**

Windows 10, Python version 3, working on Pycharm connected to Jupyter Notebook server 

- **Describe the issue that you are experiencing**

My data is from 2-photon microscopy. I both tried with an individual .tiff stack and with several files out of a dataset of 97 files. Each stack has a FOV of 301x300 pixels, 16-bit, 19 MB. They differ in the number of frames: on average each file has 100 frames. Final error message is ValueError: Error: images must really be same size for register_translation, no matter how many files I load. I am reporting the errorLog which I obtained from loading a single stack file. I am currently running demo_pipeline.ipynb.

- **Copy error log below**

---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-19-8430dfe6faf7> in <module>()
      1 #%% Run piecewise-rigid motion correction using NoRMCorre
----> 2 mc.motion_correct_pwrigid(save_movie=True)
      3 m_els = cm.load(mc.fname_tot_els)
      4 bord_px_els = np.ceil(np.maximum(np.max(np.abs(mc.x_shifts_els)),
      5                                  np.max(np.abs(mc.y_shifts_els)))).astype(np.int)  

~\CaImAn\caiman\motion_correction.py in motion_correct_pwrigid(self, save_movie, template, show_template)
    291         if template is None:
    292             print('generating template by rigid motion correction')
--> 293             self = self.motion_correct_rigid()
    294             self.total_template_els = self.total_template_rig.copy()
    295         else:

~\CaImAn\caiman\motion_correction.py in motion_correct_rigid(self, template, save_movie)
    241                 gSig_filt=self.gSig_filt,
    242                 use_cuda=self.use_cuda,
--> 243                 border_nan=self.border_nan)
    244             if template is None:
    245                 self.total_template_rig = _total_template_rig

~\CaImAn\caiman\motion_correction.py in motion_correct_batch_rigid(fname, max_shifts, dview, splits, num_splits_to_process, num_iter, template, shifts_opencv, save_movie_rigid, add_to_movie, nonneg_movie, gSig_filt, subidx, use_cuda, border_nan)
   2250                                                                  fname)[-1][:-4] + '_rig_', subidx = subidx,
   2251                                                              num_splits=num_splits_to_process, shifts_opencv=shifts_opencv, nonneg_movie=nonneg_movie, gSig_filt=gSig_filt,
-> 2252                                                              use_cuda=use_cuda, border_nan=border_nan)
   2253 
   2254         new_templ = np.nanmedian(np.dstack([r[-1] for r in res_rig]), -1)

~\CaImAn\caiman\motion_correction.py in motion_correction_piecewise(fname, splits, strides, overlaps, add_to_movie, template, max_shifts, max_deviation_rigid, newoverlaps, newstrides, upsample_factor_grid, order, dview, save_movie, base_name, subidx, num_splits, shifts_opencv, nonneg_movie, gSig_filt, use_cuda, border_nan)
   2587         print('** Finished parallel motion correction **')
   2588     else:
-> 2589         res = list(map(tile_and_correct_wrapper, pars))
   2590 
   2591     return fname_tot, res

~\CaImAn\caiman\motion_correction.py in tile_and_correct_wrapper(params)
   2443                                                                        max_deviation_rigid=max_deviation_rigid,
   2444                                                                        shifts_opencv=shifts_opencv, gSig_filt=gSig_filt,
-> 2445                                                                        use_cuda=use_cuda, border_nan=border_nan)
   2446         shift_info.append([total_shift, start_step, xy_grid])
   2447 

~\CaImAn\caiman\motion_correction.py in tile_and_correct(img, template, strides, overlaps, max_shifts, newoverlaps, newstrides, upsample_factor_grid, upsample_factor_fft, show_movie, max_deviation_rigid, add_to_movie, shifts_opencv, gSig_filt, use_cuda, border_nan)
   1882     # compute rigid shifts
   1883     rigid_shts, sfr_freq, diffphase = register_translation(
-> 1884         img, template, upsample_factor=upsample_factor_fft, max_shifts=max_shifts, use_cuda=use_cuda)
   1885 
   1886     if max_deviation_rigid == 0:

~\CaImAn\caiman\motion_correction.py in register_translation(src_image, target_image, upsample_factor, space, shifts_lb, shifts_ub, max_shifts, use_cuda)
   1408     # images must be the same shape
   1409     if src_image.shape != target_image.shape:
-> 1410         raise ValueError("Error: images must really be same size for "
   1411                          "register_translation")
   1412 

ValueError: Error: images must really be same size for register_translation

- **If you're not reporting an error, type your message below**

Thank you very much for the support!

**Comment by epnev** [comment]
What is the output of
```
sizes = [cm.load(f).shape for f in fname]
```
where `fname` is the list of files to be processed?

**Comment by Rack95** [comment]
This is the output [(108, 301, 300)]

**Comment by epnev** [comment]
I don't understand, I thought you had multiple files.

**Comment by Rack95** [comment]
I get the same error both loading a single file and multiple files. The error I posted was related to a single file. If I run with multiple files (e.g. n=10) using load_movie_chain instead of load, printing the variable you asked give as a result (1088, 301, 300). 

On the other hand, I still get:

**Comment by epnev** [comment]
I think the error is because your files are too short. Can you save these 10 files into a single file and then run it on that one? 
You can use
```
m_orig = cm.load_movie_chain(fname)
m_orig.save('new_file.h5')
```

**Comment by epnev** [comment]
@Rack95 Actually, can you try on the original short file, but by setting the parameters
`splits_els` and `splits_rig` to small number, e.g. equal to 2?

**Comment by Rack95** [comment]
Setting split_els = 2 and splits_rig = 2 the exit code is zero, so I get no error. 
Unluckily, I can't seem to make the plots work now with notebook. This is the warning I get everytime I encounter a plot line. If it's not related to this error, I can open a new issue. 


C:\Users\Rachele\Anaconda3\envs\caiman\lib\site-packages\numpy\core\_methods.py:138: RuntimeWarning: invalid value encountered in sqrt
  ret = um.sqrt(ret, out=ret)
C:\Users\Rachele\CaImAn\caiman\summary_images.py:206: RuntimeWarning: invalid value encountered in true_divide
  w_mov = (Y - np.mean(Y, axis=0)) / np.std(Y, axis=0)
C:\Users\Rachele\CaImAn\caiman\utils\visualization.py:953: UserWarning: The way to call utilities.plot_contours has changed. Look at the definition for more details.
  "Look at the definition for more details.")

On the other hand, the concatenate command to play the corrected data along side original data is shown, so motion correction and denoising is working.

**Comment by epnev** [comment]
Do you get a figure at all or not?
What's the output of `np.std(images, axis=0)`?

**Comment by Rack95** [comment]
No figures open up now when plotting, graphs are plotted directly on the IDE as shown on the image I am attaching to the comment. All figures opened up before when using the notebook. 

The output is: 

[[    0.             0.             0.         ...,     0.             0. 	
    0.        ]
 [    0.           709.1875       600.58581543 ...,  1020.39709473
    923.93292236     0.        ]
 [    0.           602.91784668   426.24310303 ...,  1013.55004883
   1091.72814941     0.        ]
 ..., 
 [    0.           486.14938354   561.67687988 ...,   813.13885498
    912.81292725     0.        ]
 [    0.           764.92480469   545.11865234 ...,   900.83843994
    734.0133667      0.        ]
 [    0.             0.             0.         ...,     0.             0.
      0.        ]]

![plot](https://user-images.githubusercontent.com/41119724/43723830-e76881e4-9998-11e8-98d3-be0e37f33e55.png)

**Comment by epnev** [comment]
The plot contours will be plot directly on the IDE. The rest of the plots are more interactive with the sliders. 
Is the problem solved now?

**Comment by Rack95** [comment]
Yes, the issue should be solved now. Thank you very much! I just have one more question about the final fluorescence results: I changed the data rate and I got no error from nb_view_patches, but still my results are not displayed. Should I check something else?

**Comment by epnev** [comment]
Glad it's resolved. 
I'm not sure whether this is a problem of the code or of windows. Can you paste the error message if any?

**Comment by Rack95** [comment]
I'm sorry for the late response. I was trying to make the pipeline run with Ubuntu on a Virtual Box just to confirm the hypothesis that it was Windows' fault, but I got memory issues. Anyway, I sticked with Windows at last and made the code run again on jupyter notebook and I was finally able to display the fluorescence signal! I conclude that there was no error with the code and that the issue is solved. Thank you very much again for the support.

**Comment by Rack95** [comment]
I'm sorry for the late response. I was trying to make the pipeline run with Ubuntu on a Virtual Box just to confirm the hypothesis that it was Windows' fault, but I got memory issues. Anyway, I sticked with Windows at last and made the code run again on jupyter notebook and I was finally able to display the fluorescence signal! I conclude that there was no error with the code and that the issue is solved. Thank you very much again for the support.
